packages/discovery-react-components/src/components/TooltipHighlight/TooltipHighlight.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { FC, useState, useEffect } from 'react';
+import cx from 'classnames';
+import { Tooltip } from 'carbon-components-react';
+import { settings } from 'carbon-components';
+import { TooltipAction, TooltipEvent } from './types';
+import { FacetInfoMap } from 'components/DocumentPreview/types';
+import { defaultMessages } from 'components/TooltipHighlight/messages';
+
+// TooltipInfo is the internal state of the TooltipHightlight
+interface TooltipInfo {
+  rectTooltipArea: DOMRect;
+  tooltipContent: JSX.Element;
+  isOpen: boolean;
+}
+
+type Props = {
+  //Parent div element
+  parentDiv: React.MutableRefObject<HTMLDivElement | null>;
+
+  // State of the highlight-tootip
+  tooltipAction: TooltipAction;
+};
+
+const baseTooltipPlaceContent = `${settings.prefix}--tooltip-place-content`;
+const baseTooltipCustomContent = `${settings.prefix}--tooltip-custom-content`;
+const baseTooltipContentHeader = `${settings.prefix}--tooltip-content-header`;
+const baseTooltipBoxColor = `${settings.prefix}--tooltip-box-color`;
+const baseTooltipContentCell = `${settings.prefix}--tooltip-content-cell`;
+const baseTooltipContentCellBuffer = `${settings.prefix}--tooltip-content-cell-buffer`;
+
+export const TooltipHighlight: FC<Props> = ({ parentDiv, tooltipAction }) => {
+  const [tooltipInfo, setTooltipInfo] = useState<TooltipInfo>({
+    rectTooltipArea: new DOMRect(),
+    tooltipContent: <div></div>,
+    isOpen: false
+  });
+
+  useEffect(() => {
+    // Compute parameters to overlay the tooltip over the active item
+    const highlightDivRect = parentDiv.current?.getBoundingClientRect() || new DOMRect();
+    const isOpen = tooltipAction.tooltipEvent !== TooltipEvent.LEAVE;
+    const clickRect = tooltipAction.rectActiveElement || new DOMRect();
+    const tooltipRect = new DOMRect(
+      clickRect.x - highlightDivRect.x,
+      clickRect.y - highlightDivRect.y,
+      clickRect?.width,
+      clickRect?.height
+    );
+    const tooltipUpdate = {
+      rectTooltipArea: tooltipRect,
+      tooltipContent: tooltipAction.tooltipContent || <div></div>,
+      isOpen: !!tooltipAction.tooltipContent && isOpen
+    };
+    setTooltipInfo(tooltipUpdate);
+  }, [tooltipAction, setTooltipInfo]);
```

Comment:
ESLint is giving me a warning here that you don't have all the dependencies added (missing `parentDiv`). If that was an intentional choice, please add a comment explaining your reasoning. If not, please add the missing var